packager/live_packager.cc

```diff
@@ -80,6 +80,39 @@ class SegmentDataReader {
   uint64_t position_ = 0;
 };
 
+class MultiSegmentDataReader {
+ public:
+  MultiSegmentDataReader(const Segment& init_segment,
+                         const Segment& media_segment)
+      : init_segment_(init_segment), media_segment_(media_segment) {}
+
+  uint64_t Read(void* buffer, uint64_t size) {
+    if (init_position_ < init_segment_.Size()) {
+      const uint64_t first_chunk_size =
+          std::min(size, init_segment_.Size() - init_position_);
+      memcpy(buffer, init_segment_.Data() + init_position_, first_chunk_size);
+
+      init_position_ += first_chunk_size;
+      return first_chunk_size;
+    } else {
+      if (position_ >= media_segment_.Size()) {
+        return 0;
+      }
+      const uint64_t second_chunk_size =
+          std::min(size, media_segment_.Size() - position_);
+      memcpy(buffer, media_segment_.Data() + position_, second_chunk_size);
+      position_ += second_chunk_size;
+      return second_chunk_size;
+    }
+  }
+
+ private:
+  const Segment& init_segment_;
+  const Segment& media_segment_;
+  uint64_t init_position_ = 0;
```

Comment:
I think you can implement this functionality with only one position. (but maybe I am missing something)
```
if (pos < init.size()) {
 // return data from the init segment
}

if(pos > init.size()) {
segPos = pos - init.size()
//return data from  media segment with segPos
}

```